### PR TITLE
Add return types in FileystemLoader

### PR DESCRIPTION
**Pull request by ruudk**

### CODE REVIEW

**Comment by stof** [comment]
If you improve the types in that class, I suggest also replacing `@param string|array $paths` with `@param string|string[] $paths` everywhere (otherwise, we cannot actually guarantee that the returned list is a list of strings)

**Comment by ruudk** [comment]
You're right, done!

File: src/Loader/FilesystemLoader.php [hunk 1]
```diff
@@ -58,14 +63,16 @@ public function getPaths(string $namespace = self::MAIN_NAMESPACE): array
...
+     * @return list<string>
      */
     public function getNamespaces(): array
     {
         return array_keys($this->paths);
     }
 
     /**
-     * @param string|array $paths A path or an array of paths where to look for templates
+     * @param string|list<string> $paths A path or an array of paths where to look for templates
```
Comment (stof) [inline src/Loader/FilesystemLoader.php]: The argument actually accepts `string[]`. It does not care whether the array is a list (as it only iterates over it ignoring the keys). Forcing it to be a list will only make it more annoying for projects using a SA tool supporting the `list` type (i.e. phpstan with bleeding-edge mode enabled or Psalm)

Comment (ruudk) [reply src/Loader/FilesystemLoader.php]: Fine, changed to `string[]` everywhere.

Comment (stof) [reply src/Loader/FilesystemLoader.php]: My comment applied only to arguments.

You should not have changed it for return types where we actually always return a list (and where having a more precise type will be *less* annoying for projects using SA tools)

Comment (ruudk) [reply src/Loader/FilesystemLoader.php]: But then I have to do array_values on the input side, oké ?

Comment (stof) [reply src/Loader/FilesystemLoader.php]: The input is never assigned directly. As said in my previous comment, it is only used in a `foreach` loop ignoring the keys.

Comment (ruudk) [reply src/Loader/FilesystemLoader.php]: You're right, I commented without seeing the actual code. Fixed it now. Thanks.

**Review by stof — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @ruudk.

**Pull request merged**